Core/src/main/java/put/io/black/java/core/logic/ScenarioManager.java

```diff
@@ -1,450 +1,493 @@
+                    addLineWithNumeration(((KeyNode) node).getChildren().get(i), scenarioWithNumeration, prefix + (i + 1) + ".");
+                }
+            }
+        }
+    }
+
+    /**
+     * Return a scenario in base form
+     * @return Scenario as text
+     */
+    public String getScenario() {
+        LinkedList<String> scenario = new LinkedList<>();
+        for (Node firstLevelNode : getFirstLevelNodes()) {
+            if (firstLevelNode instanceof KeyNode) {
+                if (((KeyNode) firstLevelNode).hasChildren()) {
+                    addLine(firstLevelNode, scenario);
+                }
+            } else {
+                String line = makeTabulaturePrefix(firstLevelNode.getNestingLevel()) + firstLevelNode.getLine() + "\n";
+                scenario.addLast(line);
+            }
+
+        }
+        return changeLinkedListToString(scenario);
+    }
+
+    /**
+     * Add line to specific node
+     * @param node Node
+     * @param scenarioWithNumeration Scenario list
+     */
+    private void addLine(Node node, LinkedList<String> scenarioWithNumeration) {
+        String line = makeTabulaturePrefix(node.getNestingLevel()) + node.getLine() + "\n";
+        scenarioWithNumeration.addLast(line);
+        if (node instanceof KeyNode) {
+            if (((KeyNode) node).hasChildren()) {
+                for (Node child : ((KeyNode) node).getChildren()) {
+                    addLine(child, scenarioWithNumeration);
+                }
+            }
+        }
+    }
+
+    /**
+     * Get scenario to selected nesting level
+     * @param maxNestingLevel Limit - nesting level
+     * @return Scenario to selected level
+     */
+    public String getScenario(int maxNestingLevel) {
+        LinkedList<String> scenario = new LinkedList<>();
+        for (Node firstLevelNode : getFirstLevelNodes()) {
+            if (firstLevelNode.getNestingLevel() <= maxNestingLevel) {
+                if (firstLevelNode instanceof KeyNode) {
+                    if (((KeyNode) firstLevelNode).hasChildren()) {
+                        addLine(firstLevelNode, scenario, maxNestingLevel);
+                    }
+                } else {
+                    String line = makeTabulaturePrefix(firstLevelNode.getNestingLevel()) + firstLevelNode.getLine() + "\n";
+                    scenario.addLast(line);
+                }
+            }
+        }
+        return changeLinkedListToString(scenario);
+    }
+
+    /**
+     * Add line
+     * @param node Node
+     * @param scenarioWithNumeration Scenario with numeration
+     * @param maxNestingLevel Max nesting level
+     */
+    private void addLine(Node node, LinkedList<String> scenarioWithNumeration, int maxNestingLevel) {
+        if (node.getNestingLevel() <= maxNestingLevel) {
+            String line = makeTabulaturePrefix(node.getNestingLevel()) + node.getLine() + "\n";
+            scenarioWithNumeration.addLast(line);
+            if (node instanceof KeyNode) {
```

Comment:
1. If not?  
2. In line 449 and 450 you cast node to KeyNode. You can cast this one to local variable.